README.md

```diff
@@ -1,6 +1,72 @@
 This repository is based on MosaicML's [llm-foundry](https://github.com/mosaicml/llm-foundry) under the Apache 2.0 license. We thank MosaicML for their foundational work in large language model training infrastructure!
 
-# LLM Foundry
+# Installation
+
+llm-foundry uses [uv](https://docs.astral.sh/uv/) to manage dependencies (uv is compatible with Conda environments, see the [GPU](#gpu) section for an example of how to integrate the two). First clone the repository. 
+
+```bash
+git clone https://github.com/LocalResearchGroup/llm-foundry.git
+cd llm-foundry
+```
+
+Then depending on your system, run one of the following commands to install the dependencies.
+
+### CPU
+
+```bash
+uv sync --extra dev --extra cpu
+```
+
+### GPU
+
+For Flash Attention support you'll need to install Cuda 12.4. The simplest way is to use [Miniconda](https://docs.anaconda.com/miniconda/install) to install it.
+
+```bash
+conda create -n llm-foundry python=3.12 uv cuda -c nvidia/label/12.4.1 -c conda-forge
+conda activate llm-foundry
+# This sets uv to use the active Conda environment whether using uv or uv pip commands
+# To permanently set uv to use the active Conda environment, add this to your ~/.bashrc
+export UV_PROJECT_ENVIRONMENT="$CONDA_PREFIX"
+```
+Or install [system Cuda](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/) (not recommended).
+
+With Cuda installed, then use uv to install the library:
+
+```bash
+uv sync --extra dev --extra gpu
+
+# Install flash attention if you have a Ampere (RTX 30xx series) or newer GPU
+uv sync --extra dev --extra gpu --extra flash
+```
+
+### MacOS
+
+```bash
+uv sync --extra dev
+```
+
+## Refresh
+
+If you used uv to install other projects or are attempting to update llm-foundry to use new dependencies, append `--refresh` to the uv sync command to update the uv cache.
+
+## Using the uv environment
+
+If you installed to the Conda environment, you can activate it with:
+
+```bash
+conda activate llm-foundry
+```
+
+if you used uv to create a virtual environment, you can activate it with:
+
+```bash
+uv venv
```

Comment:
I believe `uv venv` creates a uv virtual environment, and this is the default command to activate it:

```suggestion
uv venv
```

We might consider leaving this section out as it's not specific to this repo and thus perhaps more of a maintenance burden than it's worth?

Reply:
I fixed it. Thanks.